### DISCUSSION TITLE
Is there a way to induce the connected components in a graph?

### DISCUSSION THREAD

**Discussion by yunchong-song** [pyg-team/pytorch_geometric#6281] [Q&A]
Hi, is there a way to induce the connected components in a graph? for example, given a graph with 5 nodes and `edge_index = torch.tensor([[0,1,2,3], [1,0,3,2]])`, we want to get node's label like: `label = torch.tensor([0,0,1,1,2])`, is it possible in PyG? @rusty1s

**Comment by rusty1s** [comment]
I think you can use `scipy` for this, similar to what we do [here](https://github.com/pyg-team/pytorch_geometric/blob/master/torch_geometric/transforms/largest_connected_components.py).
